src/app/modules/form/admin-form/__tests__/admin-form.routes.spec.ts

```diff
@@ -0,0 +1,4621 @@
+
+      // Assert
+      expect(response.status).toEqual(403)
+      expect(response.body).toEqual({
+        message: `User ${defaultUser.email} not authorized to perform read operation on Form ${randomForm._id} with title: ${randomForm.title}.`,
+      })
+    })
+
+    it('should return 404 when form to duplicate cannot be found', async () => {
+      // Arrange
+      const invalidFormId = new ObjectId()
+
+      // Act
+      const response = await request.post(`/${invalidFormId}/adminform`).send({
+        responseMode: ResponseMode.Encrypt,
+        title: 'new duplicated form title',
+        publicKey: 'some public key',
+      })
+
+      // Assert
+      expect(response.status).toEqual(404)
+      expect(response.body).toEqual({ message: 'Form not found' })
+    })
+
+    it('should return 410 when form is already archived', async () => {
+      // Arrange
+      // Create archived form.
+      const archivedForm = await EmailFormModel.create({
+        title: 'Form already archived',
+        emails: [defaultUser.email],
+        admin: defaultUser._id,
+        status: Status.Archived,
+      })
+
+      // Act
+      const response = await request
+        .post(`/${archivedForm._id}/adminform`)
+        .send({
+          responseMode: ResponseMode.Email,
+          emails: 'anyrandomEmail@example.com',
+          title: 'cool new title',
+        })
+
+      // Assert
+      expect(response.status).toEqual(410)
+      expect(response.body).toEqual({ message: 'Form has been archived' })
+    })
+
+    it('should return 422 when user in session cannot be found in the database', async () => {
+      // Arrange
+      // Delete user after login.
+      await dbHandler.clearCollection(UserModel.collection.name)
+
+      // Act
+      const response = await request.post(`/${new ObjectId()}/adminform`).send({
+        responseMode: ResponseMode.Encrypt,
+        title: 'does not matter',
+        publicKey: 'some public key',
+      })
+
+      // Assert
+      expect(response.status).toEqual(422)
+      expect(response.body).toEqual({ message: 'User not found' })
+    })
+
+    it('should return 500 when database error occurs whilst duplicating form', async () => {
+      // Arrange
+      // Create form.
+      const formToDupe = await EmailFormModel.create({
+        title: 'Original form title',
+        emails: [defaultUser.email],
+        admin: defaultUser._id,
+      })
+
+      // Force validation error that will be returned as database error
+      // TODO(#614): Return transformMongoError instead of DatabaseError for better mongoose error handling.
+      const invalidEmailDupeParams = {
+        responseMode: ResponseMode.Email,
+        emails: 'notAnEmail, should return error',
+        title: 'cool new title',
+      }
+
+      // Act
+      const response = await request
+        .post(`/${formToDupe._id}/adminform`)
+        .send(invalidEmailDupeParams)
+
+      // Assert
+      expect(response.status).toEqual(500)
```

Comment:
should this be a 500? looks like a validation error, so should be 400 right?

Reply:
Yea, but I did not want to pollute this PR, and that was the old behaviour. Do you think I should just fix it here?

Reply:
I see, nah it seems out of scope